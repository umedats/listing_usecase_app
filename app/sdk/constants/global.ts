const APP_NAME = "SellingPartnerAPIListingSampleApplication";
const APP_VERSION = "1.0";
const APP_LANGUAGE = "TypeScript";
const APP_LANGUAGE_VERSION = "5.0";
/**
 * Header data to be passed to all the SP-API.
 */
export const HEADER_DATA = {
  "User-Agent": `${APP_NAME}/${APP_VERSION} (Language=${APP_LANGUAGE}/${APP_LANGUAGE_VERSION})`,
};
