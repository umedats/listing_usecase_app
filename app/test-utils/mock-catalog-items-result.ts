export const MOCK_CATALOG_ITEMS_API_RESPONSE: object = {
  numberOfResults: 1,
  items: [
    {
      asin: "B00004SPEC",
      productTypes: [
        {
          marketplaceId: "ATVPDKIKX0DER",
          productType: "KITCHEN",
        },
      ],
      summaries: [
        {
          marketplaceId: "ATVPDKIKX0DER",
          adultProduct: false,
          autographed: false,
          brand: "KRUPS",
          browseClassification: {
            displayName: "Countertop Blenders",
            classificationId: "289915",
          },
          itemClassification: "BASE_PRODUCT",
          itemName: "Krups 243-70 Power X Plus Combi Blender, DISCONTINUED",
          manufacturer: "KRUPS",
          memorabilia: false,
          modelNumber: "243-70",
          packageQuantity: 1,
          partNumber: "243-70",
          style: "Modern",
          tradeInEligible: false,
          websiteDisplayGroup: "kitchen_display_on_website",
          websiteDisplayGroupName: "Kitchen",
        },
      ],
    },
  ],
};

export const MOCK_CATALOG_ITEMS_API_EMPTY_RESPONSE: object = {
  numberOfResults: 0,
};
