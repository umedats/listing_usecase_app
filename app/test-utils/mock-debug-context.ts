import { SPAPIRequestResponse } from "@/app/model/types";

export const MOCK_SP_API_RESPONSE: SPAPIRequestResponse = {
  apiName: "GetListingsItem",
  apiDocumentationLink: "https://amazon.com",
  request: {
    marketplaceId: "ATVPDKIKX0DER",
    locale: "en_US",
    asin: "A1F83G8C2ARO7P",
  },
  response: {
    item: "Dummy Item",
  },
};
