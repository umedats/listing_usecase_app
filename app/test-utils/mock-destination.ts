import {
  CreateDestinationResponse,
  Destination,
} from "@/app/sdk/notifications";

export const MOCK_DESTINATION: Destination = {
  name: "TestDestination",
  destinationId: "5fe08e61-c5ec-4427-9f25-24a7b630cce2",
  resource: {
    eventBridge: {
      name: "sellingpartnerapi.amazon.com/amzn1.sellerapps.app.15a75829-cd4a-4efc-b947-0cc39d874577",
      region: "us-east-1",
      accountId: "123456",
    },
  },
};

export const MOCK_CREATE_DESTINATION_RESPONSE: CreateDestinationResponse = {
  payload: MOCK_DESTINATION,
  errors: {},
};
