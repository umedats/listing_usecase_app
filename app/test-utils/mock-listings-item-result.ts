export const MOCK_DELETE_LISTINGS_ITEM_API_SUCCESS_RESPONSE = {
  sku: "My SKU",
  status: "ACCEPTED",
  submissionId: "Some UUID",
};

export const MOCK_GET_LISTINGS_ITEM_API_SUCCESS_RESPONSE = {
  sku: "S014-A03",
  summaries: [
    {
      marketplaceId: "ATVPDKIKX0DER",
      asin: "B003VDOHBG",
      productType: "SUNGLASSES",
      conditionType: "new_new",
      status: [],
      itemName:
        "SKAGEN DENMARK Unisex Sunglasses Steel Aviator Frame #S014-A03",
      createdDate: "2023-02-10T01:29:04.021Z",
      lastUpdatedDate: "2023-02-10T01:29:09.595Z",
      mainImage: {
        link: "https://m.media-amazon.com/images/I/41cVJ8Gz6TL.jpg",
        height: 392,
        width: 444,
      },
    },
  ],
  attributes: {
    color: [
      {
        language_tag: "en_US",
        value: "Two-Tone",
        marketplace_id: "ATVPDKIKX0DER",
      },
    ],
  },
};

export const MOCK_GET_LISTINGS_ITEM_API_EMPTY_SUMMARY_RESPONSE = {
  sku: "S014-A03",
  summaries: [],
  attributes: {},
};

export const MOCK_PUT_AND_PATCH_LISTINGS_ITEM_API_SUCCESS_RESPONSE = {
  sku: "My SKU",
  status: "ACCEPTED",
  submissionId: "Some UUID",
};
