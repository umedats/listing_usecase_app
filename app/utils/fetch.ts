import { SPAPIRequestResponse } from "@/app/model/types";
import { SettableDebugState } from "@/app/context/debug-context-provider";
import { NO_CACHING_DIRECTIVE } from "@/app/constants/global";

/**
 * Helper function to fetch the data from the backend API. This function
 * extracts the debug context, and invokes the updateDebugContextCallback
 * method.
 * @param url the url of the backend API.
 * @param updateDebugContextCallback a callback to update the debug context.
 * @param responseDataCallback an optional call back which is invoked with the
 * data extracted from the backend API.
 * @param requestInit params to initialize the fetch request.
 */
export async function fetcherForBackendAPI(
  url: string,
  updateDebugContextCallback: (debugData: any) => void,
  responseDataCallback?: (data: any) => any | void,
  requestInit?: RequestInit,
) {
  let newRequestInit = requestInit;
  if (requestInit) {
    newRequestInit = { ...requestInit, cache: NO_CACHING_DIRECTIVE };
  }

  const response = await fetch(url, newRequestInit);
  const responseBody = await response.json();
  if (responseBody.debugContext) {
    updateDebugContextCallback(responseBody.debugContext);
  }

  if (!response.ok) {
    throw [response.status, response.statusText];
  }

  const data = responseBody.data;
  if (responseDataCallback) {
    return responseDataCallback(data);
  }
  return data;
}

/**
 * Saves the given debugData for the given route into debugContext.
 * @param pathname
 * @param debugData
 * @param debugContext
 */
export function updateDebugContext(
  pathname: string,
  debugData: any,
  debugContext: SettableDebugState,
) {
  (debugData as SPAPIRequestResponse[]).forEach((value) =>
    debugContext.addRequestResponse(pathname, value),
  );
}
