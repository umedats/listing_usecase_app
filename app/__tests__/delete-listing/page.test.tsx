import Page from "@/app/[locale]/delete-listing/page";
import { cleanup, render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@testing-library/react";
import { NextIntlProvider } from "next-intl";
import translations from "@/app/internationalization/translations/en-US.json";
import {
  mockRejectFetchResponse,
  mockResolveFetchResponse,
} from "@/app/test-utils/mock-fetch";
import userEvent from "@testing-library/user-event";
import fetch from "jest-fetch-mock";
import { MOCK_DELETE_LISTINGS_ITEM_API_SUCCESS_RESPONSE } from "@/app/test-utils/mock-listings-item-result";
import { MOCK_SP_API_RESPONSE } from "@/app/test-utils/mock-debug-context";
import { US_LOCALE } from "@/app/constants/global";

function renderPage() {
  const { asFragment, queryByText, getByText } = render(
    <NextIntlProvider locale={US_LOCALE} messages={translations}>
      <Page />
    </NextIntlProvider>,
  );

  return { asFragment, queryByText, getByText };
}

describe("Tests for the Delete Listing Page", () => {
  beforeEach(() => {
    fetch.resetMocks();
  });

  afterEach(() => {
    // Cleanup any resources or state after each test
    cleanup();
  });

  test("Snapshot Test", async () => {
    const { asFragment } = renderPage();
    expect(asFragment()).toMatchSnapshot();
  });

  test("Delete Error Test", async () => {
    mockRejectFetchResponse(500);
    renderPage();

    const byRole = screen.getByRole("textbox");
    await userEvent.type(byRole, "SKU_testtest");

    await userEvent.click(screen.getByText("Delete this listing"));

    await waitFor(() => {
      const dialog = screen.queryByText("Are you sure", { exact: false });
      expect(dialog).not.toBeNull();
      expect(dialog).toBeVisible();
    });

    expect(screen.queryByTestId("alertDialog")).toMatchSnapshot();

    await userEvent.click(screen.getByTestId("confirmAlertDialog"));

    await waitFor(() => {
      const dialog = screen.queryByText("Please try again", { exact: false });
      expect(dialog).not.toBeNull();
      expect(dialog).toBeVisible();
    });

    expect(screen.queryByTestId("alertDialog")).toMatchSnapshot();

    await userEvent.click(screen.getByTestId("cancelAlertDialog"));
    await waitFor(() => expect(screen.queryByTestId("alertDialog")).toBeNull());
  });

  test("Delete Success Test", async () => {
    mockResolveFetchResponse(200, {
      data: MOCK_DELETE_LISTINGS_ITEM_API_SUCCESS_RESPONSE,
      debugContext: [MOCK_SP_API_RESPONSE],
    });
    renderPage();

    const byRole = screen.getByRole("textbox");
    await userEvent.type(byRole, "SKU_testtest");

    await userEvent.click(screen.getByText("Delete this listing"));

    await waitFor(() => {
      const dialog = screen.queryByText("Are you sure", { exact: false });
      expect(dialog).not.toBeNull();
      expect(dialog).toBeVisible();
    });

    expect(screen.queryByTestId("alertDialog")).toMatchSnapshot();

    await userEvent.click(screen.getByTestId("confirmAlertDialog"));

    await waitFor(() => {
      const dialog = screen.queryByText("Success", { exact: false });
      expect(dialog).not.toBeNull();
      expect(dialog).toBeVisible();
    });

    expect(screen.queryByTestId("alertDialog")).toMatchSnapshot();

    await userEvent.click(screen.getByTestId("cancelAlertDialog"));
    await waitFor(() => expect(screen.queryByTestId("alertDialog")).toBeNull());
  });

  test("Delete Cancel Test", async () => {
    renderPage();

    const byRole = screen.getByRole("textbox");
    await userEvent.type(byRole, "SKU_testtest");

    await userEvent.click(screen.getByText("Delete this listing"));

    await waitFor(() => {
      const dialog = screen.queryByText("Are you sure", { exact: false });
      expect(dialog).not.toBeNull();
      expect(dialog).toBeVisible();
    });

    expect(screen.queryByTestId("alertDialog")).toMatchSnapshot();

    await userEvent.click(screen.getByTestId("cancelAlertDialog"));
    await waitFor(() => expect(screen.queryByTestId("alertDialog")).toBeNull());
  });
});
