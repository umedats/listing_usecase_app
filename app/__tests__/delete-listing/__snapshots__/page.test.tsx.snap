// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Delete Listing Page Delete Cancel Test 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="alertDialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":rk:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        id=":rk:"
      >
        Are you sure?
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          This option cannot be undone, the SKU will be permanently deleted!
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="confirmAlertDialog"
          tabindex="0"
          type="button"
        >
          I am sure
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="cancelAlertDialog"
          tabindex="0"
          type="button"
        >
          Cancel
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Tests for the Delete Listing Page Delete Error Test 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="alertDialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":r8:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        id=":r8:"
      >
        Are you sure?
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          This option cannot be undone, the SKU will be permanently deleted!
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="confirmAlertDialog"
          tabindex="0"
          type="button"
        >
          I am sure
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="cancelAlertDialog"
          tabindex="0"
          type="button"
        >
          Cancel
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Tests for the Delete Listing Page Delete Error Test 2`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="alertDialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":r9:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        id=":r9:"
      >
        Error
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          There was unexpected error while deleting the listing. Please try again.
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="cancelAlertDialog"
          tabindex="0"
          type="button"
        >
          Close
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Tests for the Delete Listing Page Delete Success Test 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="alertDialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":re:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        id=":re:"
      >
        Are you sure?
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          This option cannot be undone, the SKU will be permanently deleted!
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="confirmAlertDialog"
          tabindex="0"
          type="button"
        >
          I am sure
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="cancelAlertDialog"
          tabindex="0"
          type="button"
        >
          Cancel
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Tests for the Delete Listing Page Delete Success Test 2`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="alertDialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":rf:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        id=":rf:"
      >
        Success
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          Submitted the listing with no issues to Amazon. The submission id is : Some UUID and the status of the submission is ACCEPTED
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="cancelAlertDialog"
          tabindex="0"
          type="button"
        >
          Close
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Tests for the Delete Listing Page Snapshot Test 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-13mqtjd-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Delete a Listing
        </h6>
      </div>
    </div>
    <form
      data-testid="deleteListingRequestSubmit"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <label
            aria-label="SKU to delete."
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
            data-mui-internal-clone-element="true"
            for="skuInput"
          >
            SKU
          </label>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
        >
          <div
            aria-label="SKU to delete."
            class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
            data-mui-internal-clone-element="true"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                id="skuInput"
                name="SKU"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <span
        aria-label="Delete the Listing that's represented by this SKU."
        class=""
        data-mui-internal-clone-element="true"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="submitButton"
          disabled=""
          id="submitButton"
          tabindex="-1"
          type="button"
        >
          Delete this listing
        </button>
      </span>
    </form>
  </div>
</DocumentFragment>
`;
